lib/site-inspector.rb

```diff
@@ -127,37 +134,213 @@ def to_json
     to_hash.to_json
   end
 
-  def to_hash
+  def http
+    details = {
+      # site-inspector's best guess
+      canonical: uri.to_s,
+      canonical_protocol: https? ? :https : :http,
+      canonical_endpoint: www? ? :www : :root,
+
```

Comment:
It's still helpful to have site-inspector's best guess about a canonical domain, even if it's not perfect -- it's used once below, to determine whether HSTS is set for "the domain" as passed in, which could be for the `www` version of a domain. 

In other words, if someone asks "Does pacer.gov use HSTS?", and `https://www.pacer.gov` has HSTS set, and `https://pacer.gov` does not, but `https://www.pacer.gov` is where people go, then the answer is Yes. (And if the HSTS is reversed, the answer is no, because HSTS only includes subdomains if explicitly specified.)